query-languages/m/m-spec-types.md

```diff
@@ -337,10 +337,10 @@ A value may be ascribed a type using the library function `Value.ReplaceType`. T
 When a value is ascribed a type, only a limited conformance check occurs:
 * The type being ascribed must be non-abstract, not nullable and compatible with the value's intrinsic (native) _primitive-type_.
 * When a custom type is being ascribed, it must match in structure with the value.
-  * For records: The type must define the same number of fields as the value. (However, field names and field types are not checked.)
+  * For records: The type must be closed, must define the same number of fields as the value and must not contain any optional fields. (However, field names and field types are not checked.)
   * For tables: The type must define the same number of columns as the value. (However, column names and column types are not checked.)
   * For functions: The type must define the same number of required parameters, as well as the same number of optional parameters, as the value. (However, parameter names, parameter types and the function return type are not checked.)
-  * For lists: As a list type always defines exactly one item type, any list type is structurally compatible with all list values.
+  * For lists: Unlike records, tables and functions, which can vary in their structure, a list's structure is always defined by a single item type, so a given list type is always structurally compatible with all other list types. (However, the item type is not checked during ascription.)
```

Comment:
I'm still not entirely sure what you're getting at here. Even with a single item type, wouldn't "type {text}" (a list type with an item type of text) be different than "type {number}"? Isn't it really that there's no way to specify the item type, and thus a list type always implicitly has an item type of "any"?

Reply:
What about this?

```suggestion
  * For lists: The value must be a list. (However, the item type is not checked during ascription.)
```

Reply:
Thanks!

Reply:
If you like the proposed change, commit it and then I can sign off.

Reply:
Well, a previous bullet already says that the type being ascribed must match the value's native type, so it might be nice not to indirectly repeat that here by saying that list values must be ascribed list types.

I just tried rewording the the rules for custom types. Do you like the new wording better?

Reply:
I think separating the info about record/table/function into two different sections is more confusing than helpful.

Reply:
Returned to one list. 

However, removed the notes about column/field/param names not being checked, as this implies that those names have no effect (which is incorrect). Instead, added a paragraph + example about how the mashup engine applies those child names during ascription (i.e. as renames).